tests/unit/models/gpflux/test_sampler.py

```diff
@@ -151,7 +152,17 @@ def test_dgp_reparam_sampler_sample_is_repeatable(
 
     sampler = DeepGaussianProcessReparamSampler(100, model)
     xs = tf.random.uniform([100, 2], minval=-10.0, maxval=10.0, dtype=tf.float64)[:, None, :]
-    npt.assert_allclose(sampler.sample(xs), sampler.sample(xs))
+
+    # also check there are no unnecessary casts from float64 to float32 or vice versa
```

Comment:
(this should be a separate test but the sampler tests are currently very slow due to training; I'll try to get them to use a session-scoped fixture in a separate PR)

Reply:
yes, that sounds good